### PR TITLE
New config

### PR DESCRIPTION
**Pull request by samuelebistoletti**
I have rewritten part of the program in order to read configurations variable from a YAML file instead of environmental variables (I used Viper library). All tests are passing on my local machine. I also fixed Dockerfile and various scripts to match the new configurations. Readme is also updated with new instructions. I open this pull requests if you want to merge my work.

Thank you

### CODE REVIEW

File: redis_store.go [hunk 1]
```diff
@@ -178,7 +178,7 @@ func (this *RedisStore) Poll(c chan []byte, queue string) error {
 				continue
 			}
 
-			message, err := redis.Bytes(consumer.Do("LPOP", queue))
+			message, err := redis.Bytes(consumer.Do("BLPOP", queue, 10))
```
Comment (AdriVanHoudt) [inline redis_store.go:181]: This seems like a change not related to the config, not a redis expert but this basically blocks for 10 seconds until it can pop right?

Comment (samuelebistoletti) [inline redis_store.go:181]: Sorry, I forgot to tell you about this change. A few days ago I noticed that Incus was flooding Redis with a lot of pop requests so I made this change to improve efficiency. And yes, you are right, this way Incus will not continuously try to pop the list if it is empty.

Comment (AdriVanHoudt) [inline redis_store.go:181]: seems like a reasonable change but might be better off in a separate PR depending on the author ofc

Comment (briankassouf) [inline redis_store.go:181]: @samuelebistoletti Could you put this in a different PR? Also this change is related to https://github.com/Imgur/incus/issues/53

Comment (jacob-pascual) [inline redis_store.go:181]: This is great. Solves #53. The `time.Sleep` call below is probably unnecessary now.

**Comment by jacob-pascual** [comment]
@samuelebistoletti I am going to close this PR in favor of #61 so that you can split off a PR for the `BLPOP` change which we are not comfortable merging into master yet.

**Comment by samuelebistoletti** [comment]
it's ok. I have to reopen another PR for the BLPOP change or will you do that when you feel confortable to merge that change?

Besides I would like to suggest you to push a new image on Docker hub based on the new docker file, I successfully built and tested it on my pc, but I was not able to push it on the repo.

**Comment by druzn3k** [comment]
@jacobgreenleaf I don't understand why BLPOP has been reverted in favor of LPOP. [BLPOP](http://redis.io/commands/blpop) has been around for a long time (2.0) and proven very stable and is lighter on the CPU than the LPOP hammering and is almost a drop-in replacement for LPOP (plus the timeout parameters of course). That and it almost guarantee that what is pushed in the list is popped as soon as possibile, reducing latencies (not that redis is slow at all, but working at your scale I think that every millisecond shaved in the long run is really good). I'm really interested in what is the rationale on removing the BLPOP from this PR (not trying to be a dick, just trying to understand). Was is that the timeout is not in the config file?

**Comment by AdriVanHoudt** [comment]
@druzn3k the reason is that this PR was for the re-do of the config, changing logic like this (which in this case is a good thing) would get lost in the PR, it is cleaner to do a separate PR for it. Even you can do it ;)

**Comment by druzn3k** [comment]
Ah, ok! :+1: I thought that was for the command per-se, not for a logical separation of changes. My bad. :)
